pkg/controller/nodemaintenance/nodemaintenance_controller.go

```diff
@@ -297,9 +330,18 @@ func (r *ReconcileNodeMaintenance) stopNodeMaintenanceImp(node *corev1.Node ) er
 	return nil
 }
 
-func (r *ReconcileNodeMaintenance) stopNodeMaintenance(nodeName string) error {
+func (r *ReconcileNodeMaintenance) stopNodeMaintenanceOnDeletion(nodeName string) error {
 	node, err := r.fetchNode(nodeName)
 	if err != nil {
+		// if CR is gathered as result of garbage collection: the node may have been deleted, but the CR has not yet been deleted, still we must clean up the lease!
+		if errors.IsNotFound(err) {
+			if r.isLeaseSupported {
```

Comment:
I would normally expect that check to happen inside invalidateLease()

Reply:
agree, but probably out of scope of this PR

Reply:
I'd rather not introduce it wrong in the first place :)

Reply:
Yes, but is was already introduced, it's not a problem of this RP IMHO.